client/src/actions/publicActions.js

```diff
@@ -40,7 +40,9 @@ export const setEntityDetails = (
   payload: entityDetails,
   reducer: (state) => ({
     ...state,
-    ...entityDetails,
+    ...keyBy(map(entityDetails, (e, key) => ({...e, eid: Number.parseInt(key, 10)})), (e) =>
```

Comment:
Added eid, because it was declared in NewEntityDetails type, but was not in the object returned by Api (only in the key).